### DISCUSSION TITLE
How to test sctp load-balancing with loxilb ?

### DISCUSSION THREAD

**Discussion by CloudyBytes10** [loxilb-io/loxilb#143] [Q&A] [answered]
Hello All,

I was able to do some basic sctp load-balancer test with loxilb using tools like socat and sctp_test. Is there a way to check sctp performance ? iperf in ubuntu20.04 does not seem to support sctp option. Appreciate any pointers regarding this.

**Comment by TrekkieCoder** [comment]
It is possible to use iperf3 but it is not reliable/stable as per this [issue](https://github.com/esnet/iperf/issues/620)

**Comment by UltraInstinct14** [comment, marked as answer]
There are various open-source tools to test sctp as OP mentioned.  Just a recap for interested readers :

**1. sctp_test  (Part of lksctp-tools in Ubuntu)**

Server usage :
```
sctp_test -H 32.32.32.1 -P 5001 -l
```
* -H server_IP, -P listening port

Client usage :
```
sctp_test -H 100.100.100.1  -h 32.32.32.1 -p 5001 -s -c 1 -M 100
```
* -H local_IP, -h server_IP, -p target port

**2. socat tool**

Server usage :
```
socat sctp-listen:5001,fork STDOUT
```

Client usage :
```
echo "Hello" | socat - sctp:20.20.20.1:5001
```

**3. iperf3 tool**

Server usage :
```
iperf3 -s -p 8080 
```

Client usage :
```
iperf3 -c 88.88.88.1 -i1 -t10 --sctp
```
* iperf3 is reportedly unsable for sctp testing
